### DISCUSSION TITLE
Choosing a subscriber before rendering view

### DISCUSSION THREAD

**Discussion by ednv** [symfony/symfony#50243] [Q&A] [answered]
Hi there.

I am developing a website containing at least 200 return $this->render('template.twig', $param_array) method. This is working good but now, i'm facing an issue because i now must render a JsonResponse aside each $this->render() in my project.

So I'd like to "intercept" the render() method with a subscriber or anything else and returning a JsonResponse only with $param_array depending of an parameter in the query string.

I can do it manually with all the 200 methods by manually creating another route and returning json only in those cases, but i'm sure a subscriber would be able to handle it more easily.

Any help would be great !

**Comment by stof** [comment, marked as answer]
A subscriber won't be able to intercept calls to a PHP method.

What you could do instead is have each controller return something else than the response (an object containing your context and the template name for instance) and use a listener for the `Symfony\Component\HttpKernel\Event\ViewEvent` that will be responsible for transforming that object into a response.

**Reply by ednv** [reply]
Thanks for the idea. I'm going to try.
